Paper:  MPI-parallel Molecular Dynamics Trajectory Analysis with the H5MD Format in the MDAnalysis Python Package

edisj opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@stargaser<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@anirudhacharya, @kdpenner<!--end-reviewers-list-->

Commit:
testing commit

orbeckst committed:
authors and acknowledgements

orbeckst committed:
add people to ack and change citation format

orbeckst committed:
initial bib file

orbeckst committed:
working citations; updated acknowledgements

Commit:
updated pdf

orbeckst committed:
add 2011 MDA paper

Commit:
methods added

edisj committed:
changed name

Commit:
t_comp error

Commit:
Merge branch '2021' of https://github.com/Becksteinlab/scipy_proceedings into 2021

Commit:
longtable fix?

Commit:
longtable fix?

Commit:
removed table link

Commit:
latex \

Commit:
latex \

edisj committed:
updated pdf

Commit:
figures

edisj committed:
updated pdf

Commit:
discussion updates

edisj committed:
updated pdf

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

deniederhut commented:
@scoobies assign @stargaser as editor

scoobies commented:
Assigned! @stargaser is now the editor

anirudhacharya reviewed (commented):
@edisj I have been asked by the scipy proceedings committee to review this paper. I will do so in the coming days. I noticed a few TODOs in the paper, can you please update and complete the paper at the earliest, so that I can submit my review.

stargaser commented:
@scoobies assign @anirudhacharya as reviewer 1

scoobies commented:
Reviewer 1 assigned!

deniederhut commented:
I think the syntax is supposed to be 
@scoobies add @anirudhacharya to reviewers. I'll disable the reviewer 1
reviewer 2 thing when I get home

On May 30, 2021, ivan-marroquin ***@***.***> wrote:
> Reviewer 1 assigned!
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/scipy-
conference/scipy_proceedings/pull/639#issuecomment-851037448>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/ABUM3LDVKCUKWH5NME2SLRTTQJ4BDANCNFSM45X7Q56Q>.

deniederhut commented:
@scoobies add @anirudhacharya to reviewers

scoobies commented:
@anirudhacharya added to the reviewers list!

Commit:
changes and figures

Commit:
removed table ref

Commit:
forgot }

orbeckst committed:
fixed error in table caption

orbeckst committed:
Merge branch '2021' of github.com:Becksteinlab/scipy_proceedings into 2021

Commit:
change \_ to -

Commit:
removed another _

Commit:
remove _ from fig labels

Commit:
renamed figs

Commit:
N_processes to N processes

Commit:
testing bug fix

Commit:
testing bug fix

orbeckst committed:
edited Results: Vanilla

orbeckst committed:
Merge branch '2021' of github.com:Becksteinlab/scipy_proceedings into 2021

Commit:
moved table caption line

orbeckst committed:
fixed some instances of buried backtics

Commit:
fixed fig ref

orbeckst committed:
Merge branch '2021' of github.com:Becksteinlab/scipy_proceedings into 2021

orbeckst committed:
fixed some refs

orbeckst committed:
fixed last "literal start-string without end-string."

orbeckst committed:
use replacement role for N_processes

orbeckst committed:
edited Abstract

- updated number of coordinate file formats (also in Introduction)
- minor edits throughout (define MD first, write I/O instead of IO, ...)

orbeckst committed:
add Pierre to Ack

edisj commented:
Hi @anirudhacharya,
Apologies for the delay in having a full draft ready for you. We have a 1st draft ready now. We will likely be editing/adding details in the coming days as well but please review at your leisure

thanks,
Edis

orbeckst committed:
numpy citation added

orbeckst committed:
more refs for introduction

orbeckst committed:
fixed textbf formatting in caption

Commit:
remove white space and tabs from table

orbeckst committed:
ref for split-apply-combine

orbeckst committed:
Merge branch '2021' of github.com:Becksteinlab/scipy_proceedings into 2021

orbeckst committed:
fixed refs

edisj commented:
@deniederhut Is the latest version of our draft building on the build server? It doesn't seem to be updated

orbeckst committed:
updated acknowledgements, link to MDA, H5MD, HDF5, Lustre, BeeGFS, emacs auto mode line

orbeckst committed:
made table look better

Use booktabs and remove distracting lines

orbeckst committed:
explained and referenced RMSD

deniederhut commented:
@edisj you can click the circly-arrows button on the right side of your paper card on the build server webpage, or you can ask scoobies to do it for you with:

@scoobies build paper

deniederhut commented:
@scoobies add @kdpenner to reviewers

scoobies commented:
@kdpenner added to the reviewers list!

orbeckst committed:
Merge branch 'scipy-conference:2021' into 2021

orbeckst committed:
updated Introduction

- added references
- reduced use of adjectives
- simplified sentences

orbeckst committed:
added mpi4py ref to bib file

Commit:
methods updated with hpc details and additional test files

Commit:
remove /centering test

Commit:
correct \centering line

Commit:
updating figs - changed colors, added error bars to scaling, adding format comparison

Commit:
many updates to results, small updates to intro and conclusion, added scipy2017 paper ref

Commit:
messed up a :ref:

edisj commented:
@scoobies help

scoobies commented:
Hello @edisj, here are the things you can ask me to do:

```

# List all available commands
@scoobies help

# Show our community Code of Conduct and Guidelines
@scoobies code of conduct

# Add to this issue's reviewers list
@scoobies add @username to reviewers

# Remove from this issue's reviewers list
@scoobies remove @username from reviewers

# Assign a user as the editor of this submission
@scoobies assign @username as editor

# Remove the editor assigned to this submission
@scoobies remove editor

# Add a user to this issue's assignees list
@scoobies add assignee: @username

# Remove a user from this issue's assignees list
@scoobies remove assignee: @username

# Builds paper
@scoobies build paper

# Checks build status
@scoobies build status

# Check the references of the paper for missing DOIs
# Optionally, it can be run on a non-default branch 
@scoobies check references
@scoobies check references from custom-branch-name

# Label issue with: paper
@scoobies mark paper

# Label issue with: needs-review. Remove labels: unready, needs-more-review, pending-comment, ready
@scoobies mark needs review

# Label issue with: needs-more-review. Remove labels: unready, needs-review, pending-comment, ready
@scoobies mark needs more review

# Label issue with: pending-comment. Remove labels: unready, needs-review, needs-more-review, ready
@scoobies mark pending comment

# Label issue with: ready. Remove labels: unready, needs-review, needs-more-review, pending-comment
@scoobies mark ready

# Label issue with: unready. Remove labels: needs-review, needs-more-review, pending-comment, ready
@scoobies mark not ready

# Label issue with: does_not_build:server
@scoobies mark server fail

# Remove labels: does_not_build:server
@scoobies mark server success

```

edisj commented:
@scoobies build status

scoobies commented:
{"edisj-2021"=>{"data"=>{"build_output"=>"Invalid paper"}, "status"=>"fail"}}

deniederhut commented:
Hmm sorry that command doesn't seem to be doing the right thing. Here is the actual output:

```
[*] Updating proceedings build tools in {self.master_repo_path}...
Already up to date.
[*] Check out paper repository...
Cloning into '/tmp/tmpzioto0u4'...
Moving proceedings build tools to temp directory...
[*] Build the paper...
Building: edis_jakupovic
*** Warning: /tmp/tmpzioto0u4/publisher/../output/edis_jakupovic/paper_stats.json does not exist.
Error compiling BiBTeX
bibtex error output:
================================================================================
b'This is BibTeX, Version 0.99d (TeX Live 2017/Debian)\nThe top-level auxiliary file: paper.aux\nThe style file: alphaurl.bst\nDatabase file #1: references.bib\nWarning--I didn\'t find a database entry for "Huggins:2019"\nWarning--I didn\'t find a database entry for "Fox:2019"\nWarning--I didn\'t find a database entry for "xsede"\nWarning--I didn\'t find a database entry for "Cheatham:2015"\nWarning--I didn\'t find a database entry for "Beckstein:2018"\nWarning--I didn\'t find a database entry for "Michaud-Agrawal:2011"\nWarning--I didn\'t find a database entry for "Gowers:2016"\nWarning--I didn\'t find a database entry for "Fan:2019"\nWarning--I didn\'t find a database entry for "Khoshlessan:2020"\nWarning--I didn\'t find a database entry for "Buyl:2014"\nWarning--I didn\'t find a database entry for "Collette:2014"\nWarning--I didn\'t find a database entry for "Wickham:2011"\nWarning--I didn\'t find a database entry for "Mura:2014"\nWarning--I didn\'t find a database entry for "Harris:2020"\nWarning--I didn\'t find a database entry for "Theobald:2005vn"\nWarning--I didn\'t find a database entry for "Khoshlessan:2017"\n(There were 16 warnings)\nThis is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=pdflatex)\n restricted \\write18 enabled.\nentering extended mode\n(./paper.tex\nLaTeX2e <2017-04-15>\nBabel <3.18> and hyphenation patterns for 3 language(s) loaded.\n(./IEEEtran.cls\nDocument Class: IEEEtran 2015/08/26 V1.8b by Michael Shell\n-- See the "IEEEtran_HOWTO" manual for usage information.\n-- http://www.michaelshell.org/tex/ieeetran/\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ppl.fd)\n-- Using IEEE Computer Society mode.\n-- Using 8.5in x 11in (letter) paper.\n-- Using PDF output.\n-- This is a 10 point document.\n) (/usr/share/texlive/texmf-dist/tex/latex/cmap/cmap.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ppl.fd)<<t1.cmap>>)\n(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsmath.sty\nFor additional information on amsmath, use the `?\' option.\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amstext.sty\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsgen.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsbsy.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsopn.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/booktabs/booktabs.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/caption/caption.sty\n(/usr/share/texlive/texmf-dist/tex/latex/caption/caption3.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty))\n\nPackage caption Warning: Unsupported document class (or package) detected,\n(caption)                usage of the caption package is not recommended.\nSee the caption package documentation for explanation.\n\n) (/usr/share/texlive/texmf-dist/tex/latex/float/float.sty) (./scipy.sty\n(./status.sty (./draftwatermark.sty (./everypage.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/pdftex.def)))\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg))))\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/mathptmx.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/helvet.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/courier.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amsfonts.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amssymb.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/tools/bm.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))\n(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))\n\nPackage hyperref Warning: Option `pdftex\' has already been used,\n(hyperref)                setting the option has no effect on input line 43.\n\n(/usr/share/texlive/texmf-dist/tex/latex/tools/longtable.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/multirow/multirow.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/textcomp.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.dfu)))\n(/usr/share/texlive/texmf-dist/tex/latex/upquote/upquote.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty\nStyle option: `fancyvrb\' v2.7a, with DG/SPQR fixes, and firstline=lastline fix \n<2008/02/07> (tvz))) (./paper.aux)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ptm.fd)\n-- Lines per column: 61 (exact).\n(/usr/share/texlive/texmf-dist/tex/latex/caption/ltcaption.sty)\n(/usr/share/texlive/texmf-dist/tex/context/base/mkii/supp-pdf.mkii\n[Loading MPS to PDF converter (version 2006.09.02).]\n) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))\n(./paper.out) (./paper.out)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1phv.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ztmcm.fd)<<ot1.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omlztmcm.fd)<<oml.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omsztmcm.fd)<<oms.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omxztmcm.fd)<<omx.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ptm.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/upzd.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ts1ptm.fd) (./page_numbers.tex)\n\nLaTeX Warning: Citation `Huggins:2019\' on page 1 undefined on input line 195.\n\n\nLaTeX Warning: Citation `Fox:2019\' on page 1 undefined on input line 195.\n\n\nLaTeX Warning: Citation `xsede\' on page 1 undefined on input line 195.\n\n\nLaTeX Warning: Citation `Cheatham:2015\' on page 1 undefined on input line 197.\n\n\nLaTeX Warning: Citation `Beckstein:2018\' on page 1 undefined on input line 197.\n\n\n\nLaTeX Warning: Citation `Fox:2019\' on page 1 undefined on input line 197.\n\n\nLaTeX Warning: Citation `Michaud-Agrawal:2011\' on page 1 undefined on input lin\ne 198.\n\n\nLaTeX Warning: Citation `Gowers:2016\' on page 1 undefined on input line 198.\n\n\nLaTeX Warning: Citation `Fan:2019\' on page 1 undefined on input line 199.\n\n\nLaTeX Warning: Citation `Khoshlessan:2020\' on page 1 undefined on input line 20\n0.\n\n\nLaTeX Warning: Citation `Buyl:2014\' on page 1 undefined on input line 202.\n\n\nLaTeX Warning: Citation `Collette:2014\' on page 1 undefined on input line 204.\n\n\nLaTeX Warning: Citation `Collette:2014\' on page 1 undefined on input line 206.\n\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1pcr.fd)\n\nLaTeX Warning: Citation `xsede\' on page 1 undefined on input line 209.\n\n\nLaTeX Warning: Citation `Wickham:2011\' on page 1 undefined on input line 210.\n\n\nLaTeX Warning: Citation `Mura:2014\' on page 1 undefined on input line 212.\n\n[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}]\n\nLaTeX Warning: Citation `Harris:2020\' on page 2 undefined on input line 216.\n\n\nUnderfull \\hbox (badness 4328) in paragraph at lines 260--262\n[]\\T1/ptm/m/n/9.53561 install h5py with \\T1/pcr/m/n/9.53561 CC="mpicc" HDF5_MPI\n="ON"\n\nUnderfull \\hbox (badness 10000) in paragraph at lines 260--262\n\\T1/pcr/m/n/9.53561 HDF5_DIR=/path/to/parallel-hdf5 pip\n\nOverfull \\hbox (101.78712pt too wide) in paragraph at lines 267--276\n [][] \n\nLaTeX Warning: Citation `Fan:2019\' on page 2 undefined on input line 287.\n\n\nUnderfull \\hbox (badness 1824) in paragraph at lines 286--309\n\\T1/ptm/m/n/9.53561 The test data files used in our bench-mark con-sist of a\n\nUnderfull \\hbox (badness 1939) in paragraph at lines 286--309\n\\T1/ptm/m/n/9.53561 a tra-jec-tory file \\T1/pcr/m/n/9.53561 YiiP_system_9ns_cen\nter100x.h5md\n\nLaTeX Warning: Citation `Wickham:2011\' on page 2 undefined on input line 316.\n\n\nLaTeX Warning: Citation `Fan:2019\' on page 2 undefined on input line 316.\n\n\nLaTeX Warning: Citation `Khoshlessan:2020\' on page 2 undefined on input line 31\n6.\n\n\nLaTeX Warning: Citation `Mura:2014\' on page 2 undefined on input line 318.\n\n\nLaTeX Warning: Citation `Theobald:2005vn\' on page 2 undefined on input line 318\n.\n\n\nLaTeX Warning: Citation `Fan:2019\' on page 2 undefined on input line 318.\n\n\nLaTeX Warning: Citation `Khoshlessan:2020\' on page 2 undefined on input line 31\n8.\n\n[2]\n\nLaTeX Warning: Citation `Harris:2020\' on page 3 undefined on input line 409.\n\n\nUnderfull \\hbox (badness 4621) in paragraph at lines 419--425\n\\T1/ptm/m/n/9.53561 to read the en-tire file in one go and pass all file\n\nUnderfull \\hbox (badness 2035) in paragraph at lines 419--425\n\\T1/ptm/m/n/9.53561 tion, we al-lowed \\T1/pcr/m/n/9.53561 u.load_new() \\T1/ptm/\nm/n/9.53561 to take a list or ar-\n\nUnderfull \\hbox (badness 2213) in paragraph at lines 419--425\n\\T1/ptm/m/n/9.53561 ray of atom in-dices as an ar-gu-ment, \\T1/pcr/m/n/9.53561 \nsub\\T1/ptm/m/n/9.53561 , so that the\n\nLaTeX Warning: Citation `Buyl:2014\' on page 3 undefined on input line 439.\n\n\nLaTeX Warning: Citation `Khoshlessan:2017\' on page 3 undefined on input line 44\n3.\n\n\nLaTeX Warning: Citation `Khoshlessan:2020\' on page 3 undefined on input line 44\n3.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 446.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 449.\n\n\nUnderfull \\vbox (badness 5681) has occurred while \\output is active [3]\n\nLaTeX Warning: Citation `Harris:2020\' on page 4 undefined on input line 464.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 471.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 474.\n\n[4 <./figs/components-vanilla.pdf> <./figs/scaling-vanilla.pdf\n\npdfTeX warning: pdflatex (file ./figs/scaling-vanilla.pdf): PDF inclusion: mult\niple pdfs with page group included in a single page\n>]\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 495.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 498.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 521.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 524.\n\n\nUnderfull \\vbox (badness 10000) has occurred while \\output is active [5 <./figs\n/components-masked.pdf> <./figs/scaling-masked.pdf\n\npdfTeX warning: pdflatex (file ./figs/scaling-masked.pdf): PDF inclusion: multi\nple pdfs with page group included in a single page\n> <./figs/components-mem.pdf\n\npdfTeX warning: pdflatex (file ./figs/components-mem.pdf): PDF inclusion: multi\nple pdfs with page group included in a single page\n> <./figs/scaling-mem.pdf\n\npdfTeX warning: pdflatex (file ./figs/scaling-mem.pdf): PDF inclusion: multiple\n pdfs with page group included in a single page\n>]\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 539.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 542.\n\n\nUnderfull \\vbox (badness 1242) has occurred while \\output is active\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 560.\n\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp\') on input line 563.\n\n[6 <./figs/serial-IO.pdf> <./figs/format-comparison.pdf\n\npdfTeX warning: pdflatex (file ./figs/format-comparison.pdf): PDF inclusion: mu\nltiple pdfs with page group included in a single page\n> <./figs/components-chunk.pdf\n\npdfTeX warning: pdflatex (file ./figs/components-chunk.pdf): PDF inclusion: mul\ntiple pdfs with page group included in a single page\n>] (./paper.bbl\n\n! LaTeX Error: Something\'s wrong--perhaps a missing \\item.\n\nSee the LaTeX manual or LaTeX Companion for explanation.\nType  H <return>  for immediate help.\n ...                                              \n                                                  \nl.3 \\end{thebibliography}\n                         \n!  ==> Fatal error occurred, no output PDF file produced!\nTranscript written on paper.log.\n'
================================================================================
[X] Paper build failed.
A build error occurred during the self.retrieve_pdf step.
```

Looks like an error in your bibfile. Does this compile locally for you?

Commit:
retry references

edisj commented:
I've been having trouble building locally :(, so I've been relying on the build server

Commit:
reverting bib

orbeckst committed:
updated pyh5md

- link to repo
- state that it is unmaintained
- mention @pdebuyl and our native MDA implementation

orbeckst committed:
updated software stack

- cite mpi4py
- link mpi4py
- link OpenMPI and capitalize
- capitalize C

orbeckst committed:
removed stray URLs from references

orbeckst committed:
spell checked

orbeckst committed:
introduction: reference task-based vs MPI

orbeckst committed:
polishing introduction

Commit:
transpose table so it will fit

Commit:
update figs with better ticks and axis scales

kdpenner commented:
Howdy, my name is Kyle Penner. I have a Ph.D. in astronomy; this paper is outside my domain of expertise so I hope to provide you a review that will make the paper understandable to a general technical audience. I'll have an initial review by June 10.

deniederhut commented:
We've been having difficulty finding reviewers for some papers, so to make sure everyone has enough time to do their reviews, we will be bumping the deadline for the initial review back to June 17th.

stargaser commented:
Hi  @anirudhacharya and @kdpenner, just a friendly reminder that the initial review deadline is next week.

(for the latter, I am an astronomer too!)

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+
+
+---------------------------------------------------------------------------------------------------------
+MPI-parallel Molecular Dynamics Trajectory Analysis with the H5MD Format in the MDAnalysis Python Package
```
The title seems generic and domain specific. And incomplete because you not only analyze trajectories, you emphasize file I/O. I'm partial to titles with conclusions: "A file I/O speedup of 20x from..."

orbeckst replied:
I agree with your assessment. Nevertheless, we'd like to keep the title because it indicates to the domain scientists, which will (after the conference) be the primary audience for the paper, exactly what it's about.

kdpenner replied:
@deniederhut @stargaser question for the editors: are these papers supposed to be written for folks at the conference to consult afterwards? for domain experts who didn't attend?

stargaser replied:
Since we have domain-specific tracks and mini-symposia, talks and posters can be aimed at those audiences. 

I don't have a problem with the title.

kdpenner replied:
👍

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+.. class:: abstract
+
+   MDAnalysis_ is a widely used Python library that can read and write over 25 popular molecular dynamics (MD) coordinate file formats.
+   With the increasing size of MD simulation files, parallel analysis is becoming a necessity for the efficient use of time and HPC resources.
```
order of magnitude for size of files?

edisj replied:
added

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+
+.. class:: abstract
+
+   MDAnalysis_ is a widely used Python library that can read and write over 25 popular molecular dynamics (MD) coordinate file formats.
```
why are there so many file formats?

edisj replied:
I think it just has to do with all of the programs that generate simulation files generate different formats, and there was never a real standard put in place. That being said, different formats do different things well, for instance the XTC format is highly compressed so you save on disk space but sacrifice a little I/O speed.

orbeckst replied:
Good question — history, and a general fragmentation of the eco system, with entrenched used communities. MDAnalysis was started as a project to allow scientists more easily to move between these entrenched user communities that used to aggregate around different MD packages.

kdpenner replied:
would be useful to mention the point about fragmentation and entrenched communities

edisj replied:
I mentioned it in the intro :)

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+   MDAnalysis_ is a widely used Python library that can read and write over 25 popular molecular dynamics (MD) coordinate file formats.
+   With the increasing size of MD simulation files, parallel analysis is becoming a necessity for the efficient use of time and HPC resources.
+   For any task-based approach to parallel analysis, file I/O typically becomes the limiting factor of overall analysis speed.
+   Our previous feasibility study suggested that parallel reading via MPI-IO and HDF5 can lead to good scaling, so we implemented a parallel MPI-IO capable HDF5-based file format trajectory reader into MDAnalysis, H5MDReader, that adheres to H5MD_ (HDF5 for Molecular Dynamics) specifications.
```
I don't know what MPI-IO is; had to look it up

edisj replied:
I feel like I have two options here: be more careful in defining the jargon or removing the jargon from the abstract and speak in more plain terms. I think in this case MPI-IO is commonly  understood in the HPC community that it's ok to leave it as is so people in the field can quickly see that my paper involves MPI

kdpenner replied:
fair enough; I'm not an HPC person

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+   We benchmarked its parallel file reading capabilities on three HPC clusters: ASU Agave, SDSC Comet, and PSC Bridges.
+   The benchmark consisted of a simple split-apply-combine scheme of an I/O bound task that split a 90k frame (113GB) trajectory into :math:`N` chunks for :math:`N` processes, where each process performed an RMSD calculation on their chunk of data, and then gathered the results back to the root process.
+   We found maximum I/O speedups at 2 full nodes, with Agave showing 20x, while Bridges and Comet capped out at 4-5x speedup.
+   On the other hand, the computation time of the RMSD calculation scaled well on all three HPC resources, with a maximum speedup on Comet of 373x on 384 cores.
```
please define RMSD here

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+   We found maximum I/O speedups at 2 full nodes, with Agave showing 20x, while Bridges and Comet capped out at 4-5x speedup.
+   On the other hand, the computation time of the RMSD calculation scaled well on all three HPC resources, with a maximum speedup on Comet of 373x on 384 cores.
+   Therefore, for a compute bound task, our implementation would likely scale very well although the file I/O still seems to impose a bottleneck on the total scaling for the I/O bound task.
+   To investigate MPI rank competition, we increased the stripe count on Bridge’s and Comet’s Lustre file system up to 96 and found marginal I/O scaling improvements of 1.2x on up to 4 full nodes.
```
I don't know what stripes are or what stripe count means

edisj replied:
I removed a lot of the jargon in the abstract and added some more detail on striping in the introduction

kdpenner replied:
I think I commented elsewhere, but if not, stripe count still needs to be defined

edisj replied:
Defined in the intro :D

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+Introduction
+============
+
+The molecular dynamics (MD) simulation approach :cite:`Huggins:2019` is widely used across the biomolecular and materials sciences, accounting for more than one quarter of the total computing time :cite:`Fox:2019` in the Extreme Science and Engineering Discovery Environment (XSEDE) network of national supercomputers in the US :cite:`xsede`.
```
I think you need a few sentences on what these simulations simulate and why they're important. What variables do you have to calculate? What do I not know if I don't know these variables?

edisj replied:
I added a couple sentences highlighting the benefits of MD simulations when studying protein dynamics and linked some references rather than giving explicit examples. Do you think this is enough for a general reader?

kdpenner replied:
I think a few more sentences (+ maybe an example?) are warranted. I see the word "dynamics" and think of movement and energy transfer. Are these simulations of an isolated protein moving in the presence of an electromagnetic field? How two molecules combine to form another? How do you get to drug discovery from such a simulation?

The rest of this edit is good!

edisj replied:
I added a reference citing recent work involving the Covid that I hope exemplifies a useful way MD simulations are used

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+    TRR              & TRR        & 113    \\
+    \bottomrule
+   \end{tabular}
+   \caption{Data files benchmarked on all three HPCS. \textbf{name} is the name that is used to identify the file in this paper.\textbf{format} is the format of the file, and \textbf{file size} gives the size of the file in gigabytes. \textbf{H5MD-default} original data file written with ``pyh5md`` :cite:`Buyl:2014` which uses the auto-chunking algorithm in ``h5py``. \textbf{H5MD-chunked} is the  ...
```
table header implies gibibyte, not gigabyte

Buyl citation doesn't render

edisj replied:
fixed, thanks!

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+                        root=0)
+       t_comm_gather = comm_gather.elapsed
+
+The time |tinit_top| records the time it takes to load a ``universe`` from the topology file.
+|tinit_traj| records the time it takes to open the trajectory file.
+The HDF5 file is opened with the ``mpio`` driver and the ``MPI.COMM_WORLD`` communicator to ensure the file is accessed in parallel via MPI I/O.
+It's important to separate the topology and trajectory initialization times, as the topology file is not opened in parallel and represents a fixed cost each process must pay to open the file.
+|tIO| represents the time it takes to read the data for each frame into the corresponding ``MDAnalysis.Universe.trajectory.ts`` attribute.
+MDAnalysis reads data from MD trajectory files one frame, or "snapshot" at a time.
+Each time the ``u.trajectory[frame]`` is iterated through, MDAnalysis reads the file and fills in numpy arrays :cite:`Harris:2020` corresponding to that time step.
+Each MPI process runs an identical copy of the script, but receives a unique ``start`` and ``stop`` variable such that the entire file is read in parallel.
+|tcomp| gives the total RMSD computation time.
+|twait| records how long each process waits before the results are gathered with ``comm.Gather()``.
+Gathering the results is done collectively by MPI, which means all processes must finish their iteration blocks before the results can be returned.
+Therefore, it's important to measure |twait| as it represents the existence of "straggling" processes.
+If one process takes substantially longer than the others to finish its iteration block, all processes are slowed down.
+|tcomm| measures the time MPI spends communicating the results from each process back to the root process.
+These 6 timings are returned and saved as an array for each benchmark run.
```
I suggest this paragraph be restyled as bullet points

edisj replied:
You're right, this was a little too clunky as a paragraph. I put the timings with corresponding line numbers from the code into a table instead

kdpenner replied:
great! much more readable

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,501 @@
+   Benchmark timings breakdown for the ASU Agave, PSC Bridges, and SDSC Comet HPC clusters. The benchmark was run on up to 4 full nodes on each HPC, where |NProcesses| was 1, 28, 56, and 112 for Agave and Bridges, and 1, 24, 48, and 96 on Comet. The ``H5MD-default`` file was used in the benchmark, where the trajectory was split in N chunks for each corresponding N process benchmark. Points repres ...
+   :label:`fig:components-vanilla`
+
+.. figure:: figs/scaling-vanilla.pdf
```
in the strong scaling figures the x axes for the center and right plots don't span 10^0 - 10^2 as does the x axis for the left plot

edisj replied:
The center and right plots are on a linear scale while the left plot is logarithmic. I think leaving the x axes as they are better represents each corresponding data

kdpenner replied:
why is the left plot log-log? seems like log-lin is the better choice. Now that I think about it, perhaps all of the x axes should be linear

edisj replied:
We changed all of the plots to lin-log, as you suggested. The original goal of the log-log scales was to show the scaling of the I/O and RMSD lines on the component plots, but since we have separate plots that explicitly show scaling of those two components, the linear x is a bit more readable.

kdpenner reviewed (commented):
As I mentioned in a previous comment, I don't have domain expertise. Additionally I don't have expertise in HPC or parallel computing. So my review won't be on the technical merits. General feedback:

1. The abstract includes sentences which were copy-pasted from the body of the paper, which is bad form. It's long. I suggest: 1 sentence on why these simulations are important; 1 on the challenges of the simulations, including the challenge that is the focus of the paper; 2 on what you did and how you did it; 1 on conclusions; and 1 on implications. Please write an abstract that entices me to read the rest of the paper, despite my lack of domain expertise.
2. All of the data shown in Figs 1--6 and 9--12 are better served in a table. Criss-crossing and overlapping lines make them hard to see, and a line connecting two data points is extra ink on the plot.
3. The plots I'd like to see compare methods and show relative differences. e.g. a comparison of the `masked_array` technique with the loading-into-memory technique.
4. Please expand on the implications of the work. Over an entire simulation, how much time will you save? Order of magnitude is fine. How does the time you save compare to the length of the award for HPC resources? What can you do with the saved time?

I can tell that you've done something novel. The BLUF is that the benefits of a bit of customization of a widely used Python library are worth the costs. (Right?) Just make a few modifications to the paper and you'll convince me to do so in my own work.

deniederhut commented:
@scoobies mark needs more review

edisj commented:
Hi @kdpenner, thx for the detailed review! I'll address the comments shortly

deniederhut commented:
@edisj to give everyone more time to respond to their reviews, we've pushed back the deadline for authors' final revisions to July 1st

stargaser commented:
@scoobies mark pending comment

deniederhut commented:
Hi @edisj ! Have you had a chance to respond to your reviewers' comments?

edisj commented:
Hi @deniederhu , yes! The revised version will be uploaded by the end of the day

Commit:
addressing review

edisj commented:
> As I mentioned in a previous comment, I don't have domain expertise. Additionally I don't have expertise in HPC or parallel computing. So my review won't be on the technical merits. General feedback:
> 
> 1. The abstract includes sentences which were copy-pasted from the body of the paper, which is bad form. It's long. I suggest: 1 sentence on why these simulations are important; 1 on the challenges of the simulations, including the challenge that is the focus of the paper; 2 on what you did and how you did it; 1 on conclusions; and 1 on implications. Please write an abstract that entices me to read the rest of the paper, despite my lack of domain expertise.
> 2. All of the data shown in Figs 1--6 and 9--12 are better served in a table. Criss-crossing and overlapping lines make them hard to see, and a line connecting two data points is extra ink on the plot.
> 3. The plots I'd like to see compare methods and show relative differences. e.g. a comparison of the `masked_array` technique with the loading-into-memory technique.
> 4. Please expand on the implications of the work. Over an entire simulation, how much time will you save? Order of magnitude is fine. How does the time you save compare to the length of the award for HPC resources? What can you do with the saved time?
> 
> I can tell that you've done something novel. The BLUF is that the benefits of a bit of customization of a widely used Python library are worth the costs. (Right?) Just make a few modifications to the paper and you'll convince me to do so in my own work.

Hi @kdpenner , I just pushed the revisions where I hope I've addressed all of your comments. Please let me know if you have any other comments! 

For the points listed here -
1. I tried to shorten the abstract and simplify the jargon, but it ended up being equally as long. Although I think it it focuses more on the conclusions and implications now.
2. While I agree there is a lot of data to slug through and perhaps there is an easier, more digestible way to go about it, I don't think this sort of data can simply be catalogued on a table. We're not really interested in the absolute time of these benchmark runs, as that can fluctuate drastically on a single machine (time of day, getting a bad ssd, file system traffic). So the slope of the lines as N_processes increases is more important than the actual values of the points
3. This is a good point. I added an additional plot directly comparing the three methods 
4. I tried to add more emphasis on how much time is exactly saved

orbeckst committed:
minor final fixes

- abstract: explain RMSD, add H5MDWriter
- removed some colloqialisms
- spell checked

orbeckst commented:
@kdpenner many thanks for giving our contribution a thorough read with thoughtful comments. 

@edisj I did my final edits and pushed them, well done!

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -93,9 +96,11 @@ The RMSD calculation is a very common task performed to analyze the dynamics of
 Because it is a fast computation that is bounded by how quickly data can be read from the file it is a suitable task to test the I/O capabilities of H5MDReader.
 
 Across the three HPC clusters tested, the benchmarks were performed on BeeGFS_ and Lustre_ parallel file systems, which are suited for multi-node MPI parallelization.
+One key feature of a Lustre parallel file systems is `file striping`_, which is the ability to store data from a file across multiple physical locations, known as object storage targets (OSTs).
```
striping is defined here but results are mentioned in the context of stripe count. Please expand on stripe count

kdpenner commented on papers/edis_jakupovic/edis_jakupovic.rst:
```diff
@@ -0,0 +1,515 @@
+   \end{tabular}
+   \caption{Data files benchmarked on all three HPCS.
+            \textbf{name} is the name that is used to identify the file in this paper.
+            \textbf{format} is the format of the file, and \textbf{file size} gives the size of the file in gigabytes.
```
still a discrepancy here: table header shows GiB (gibibytes), while description has GB (gigabytes)

kdpenner commented:
I think one takeaway from reviewing a paper on github is that the process can get complicated when there are lots of commits and the history is complicated. I hope all of my comments render properly.

My additional comment is that I'm still interested in an order-of-magnitude number for award length. Saving an hour of read time on a 2 day award means immediate implementation; saving an hour of read time on a 1 month award is laudable but more theoretical.

kdpenner commented:
Two more comments:

1. I recommend moving the explanation of striping to immediately after the first introduction of Lustre
2. The example you might use to explain MD simulations to the non-expert is the C_alpha one you use to calculate the RMSD

stargaser commented:
Thank you @kdpenner for checking the revisions. 

The proceedings committee reviewed the paper earlier today and is planning to include it in the pre-conference publication which will happen in about a day from now. The final publication is in late July which could give an opportunity to address the remaining points.

stargaser commented:
@scoobies mark ready

orbeckst commented:
Thank you @kdpenner ; we would like to address the remaining points before the final publication.

deniederhut commented:
Okay, I'll keep this PR open so we don't lose Kyle's comments. Edis and Oliver -- your conference-ready paper has been published online here: http://conference.scipy.org/proceedings/scipy2021/edis_jakupovic.html

edisj committed:
poster added

edisj committed:
updated figs with lin-log scale

Commit:
addressing final review points

Commit:
zenodo doi

orbeckst committed:
data sharing DOI update

use the general DOI and use the DOI resolver in the link

deniederhut commented:
Hey @edisj ! It looks like you've been pushing some updates to this paper recently. That's totally okay -- I just wanted to give you a heads-up that we'll be doing all the final merges and publication this week-end, so please have any further changes pushed before then.

edisj commented:
@kdpenner thanks again for your thorough review! I think the paper ended in a much better spot than it started with your contributions :). I hope I've addressed your final points. Also, I added a _Data Sharing_ section in the methods with links to all our benchmark and submission scripts

@deniederhut I reviewed/proofread the paper one last time.  Thanks again for this opportunity!

deniederhut commented:
Thanks for being a part of the conference! 😄

On July 30, 2021, ivan-marroquin ***@***.***> wrote:
> @kdpenner <https://github.com/kdpenner> thanks again for your thorough
> review! I think the paper ended in a much better spot than it started
> with your contributions :). I hope I've addressed your final points.
> Also, I added a Data Sharing section in the methods with links to all
> our benchmark and submission scripts
>
> @deniederhut <https://github.com/deniederhut> I reviewed/proofread the
> paper one last time. Thanks again for this opportunity!
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/scipy-
conference/scipy_proceedings/pull/639#issuecomment-890157172>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/ABUM3LHQPR7CMV2OFX2HU6LT2MI3NANCNFSM45X7Q56Q>.
